### PR TITLE
Add macros for construction of pmix_info_t lists

### PR DESCRIPTION
**Pull request by rhc54**
Greatly helpful when building pmix_info_t arrays!

Signed-off-by: Ralph Castain <rhc@pmix.org>

### CODE REVIEW

**Comment by rhc54** [comment]
@jjhursey Forgot to add these to the previous commit - I found them really helpful

File: Chap_API_Struct.tex [hunk 1]
```diff
@@ -1291,6 +1291,116 @@ \subsubsection{Info structure support macros}
...
+
+Note that the pointer will be initialized to an opaque structure whose elements are implementation-dependent. The caller must not modify or dereference the object.
+
+%%%%%%%%%%%
+\littleheader{Add a \refstruct{pmix_info_t} structure to a list}
+\declaremacro{PMIX_INFO_LIST_ADD}
+
+Add a \refstruct{pmix_info_t} structure containing the specified value to the provided list.
+
+\versionMarker{2.0}
```
Comment (jjhursey) [inline Chap_API_Struct.tex:1323]:
```suggestion
\versionMarker{4.0}
```

File: Chap_API_Struct.tex [hunk 2]
```diff
@@ -1291,6 +1291,116 @@ \subsubsection{Info structure support macros}
...
+Both key and data will be copied into the \refstruct{pmix_info_t} on the list - thus, the key and any data stored in the source value can be modified or free'd without affecting the copied data once the macro has completed.
+\adviceuserend
+
+%%%%%%%%%%%
+\littleheader{Transfer a \refstruct{pmix_info_t} structure to a list}
+\declaremacro{PMIX_INFO_LIST_XFER}
+
+Transfer the information in a \refstruct{pmix_info_t} structure to the provided list.
+
+\versionMarker{2.0}
```
Comment (jjhursey) [inline Chap_API_Struct.tex:1348]:
```suggestion
\versionMarker{4.0}
```

File: Chap_API_Struct.tex [hunk 3]
```diff
@@ -1291,6 +1291,116 @@ \subsubsection{Info structure support macros}
...
+All data (including key, value, and directives) will be copied into the destination \refstruct{pmix_info_t} on the list - thus, the source \refstruct{pmix_info_t} may be free'd without affecting the copied data once the macro has completed.
+\adviceuserend
+
+%%%%%%%%%%%
+\littleheader{Convert a \refstruct{pmix_info_t} list to an array}
+\declaremacro{PMIX_INFO_LIST_CONVERT}
+
+Transfer the information in the provided \refstruct{pmix_info_t} list to a \refstruct{pmix_data_array_t} array
+
+\versionMarker{2.0}
```
Comment (jjhursey) [inline Chap_API_Struct.tex:1371]:
```suggestion
\versionMarker{4.0}
```

File: Chap_API_Struct.tex [hunk 4]
```diff
@@ -1291,6 +1291,116 @@ \subsubsection{Info structure support macros}
...
+\cspecificstart
+\begin{codepar}
+PMIX_INFO_LIST_XFER(rc, m, d)
```
Comment (jjhursey) [inline Chap_API_Struct.tex:1374]: I think you mean `PMIX_INFO_LIST_CONVERT` here

File: Chap_API_Struct.tex [hunk 5]
```diff
@@ -1291,6 +1291,116 @@ \subsubsection{Info structure support macros}
...
+\argin{d}{Pointer to an instantiated \refstruct{pmix_data_array_t} structure where the \refstruct{pmix_info_t} array is to be stored (pointer to \refstruct{pmix_data_array_t})}
+\end{arglist}
+
+%%%%%%%%%%%
+\littleheader{Release a \refstruct{pmix_info_t} list}
+\declaremacro{PMIX_INFO_LIST_RELEASE}
+
+Release the provided \refstruct{pmix_info_t} list
+
+\versionMarker{2.0}
```
Comment (jjhursey) [inline Chap_API_Struct.tex:1390]:
```suggestion
\versionMarker{4.0}
```

File: Chap_API_Struct.tex (hunk 4 above)
Comment (rhc54) [reply Chap_API_Struct.tex:1374]: Yeah, I got sloppy with copy/paste - will fix

**Comment by rhc54** [comment]
@jjhursey Should be good to go now

File: Chap_Revisions.tex [hunk 6]
```diff
@@ -537,21 +537,30 @@ \section{Version 4.0: Nov. 2020}
...
+        \item \refapi{PMIx_tool_get_servers}
+    \end{compactitemize}
+
+    \item Macros
+    \begin{compactitemize}
+        \item \refmacro{PMIX_INFO_LIST_START}
+        \item \refmacro{PMIX_INFO_LIST_ADD}
+        \item \refmacro{PMIX_INFO_LIST_XFER}
+        \item \refmacro{PMIX_INFO_LIST_CONVERT}
+        \item \refmacro{PMIX_INFO_LIST_RELEASE}
```
Comment (jjhursey) [inline Chap_Revisions.tex:563]: The macros should be moved to the `Added Macros` section with `PMIX_CHECK_RESERVED_KEY`

**Comment by rhc54** [comment]
This is going against master instead of v4 - I'll close and reopen it against v4

**Pull request closed without merging**